opnode/node/api.go

```diff
@@ -74,3 +90,105 @@ func toBlockNumArg(number rpc.BlockNumber) string {
 	}
 	return hexutil.EncodeUint64(uint64(number.Int64()))
 }
+
+type BatchBundleRequest struct {
+	// L2History is a list of L2 blocks that are already in-flight or confirmed.
+	// The rollup-node then finds the common point, and responds with that point as PrevL2BlockHash and PrevL2BlockNum.
+	// The L2 history is read in order of the provided hashes, which may contain arbitrary gaps and skips.
+	// The first common hash will be the continuation point.
+	// A batch-submitter may search the history using gaps to find a common point even with deep reorgs.
```

Comment:
I don't think the batch-submitter does a new request with different (longer or older) history when an error is returned by the rollup node that no common chain was found with previous hashes. Can you create an issue to test deep reorgs + implement this in the batch submitter?